### DISCUSSION TITLE
Colour issues with ESP32-2432S028R (2USB variant)

### DISCUSSION THREAD

**Discussion by DynaMight1124** [moononournation/Arduino_GFX#414] [General]
Hi

Thank you for the video library and everything else you do. I am currently trying to port the MiniTV to the ESP32-2432S028R (also known as Cheap Yellow Display/CYD) but theres a new variant out with 2 USB ports, a Micro USB and also USBC. Known as CYD2USB.

With the demo graphics library (TFT_eSPI) it was noted the 2 USB version needed TFT_INVERSION_ON flag else the colours would be inverted, theres also some differences in gamma that cause a very washed out image (https://github.com/Bodmer/TFT_eSPI/issues/2985)

When I use the sample code for this display on your library I get the below display:
![PXL_20240106_104909469](https://github.com/moononournation/Arduino_GFX/assets/18290628/69896a60-7378-460f-af43-064543de56b8)

I am not sure if the garbled image on the right is related to the graphics library or the MiniTV sketch (MJPEG_BUFFER_SIZE is set to320x240 and videos also encoded at 320@30fps so should work?) anyway there main issue is the colours, its not totally inverted but opening should look like this:
![320_30fps mjpeg_snapshot_00 06 801](https://github.com/moononournation/Arduino_GFX/assets/18290628/26656627-85f2-4dc7-9f80-3e5e990feb71)

I dont know if this is enough to go on? If you need any other info please let me know.

Thanks

**Comment by moononournation** [comment]
This is not an issue, please discuss in Discussions.

**Reply by KevynWithaY** [reply]
Make sure you're selecting the correct display class. I thought my CYD2USB had an ILI9341 initially until I notice the little label on the plastic case said (chinese characters)7789. So, I tried the ST7789 display class and that works. My CYD is happily playing video now using code from the "MJPEG_2task_Audio_1task" example. Haven't tried the audio yet, as I don't have the proper cable/connector for that port.

**Reply by DynaMight1124** [reply]
Yes you're right, ST7789 does work! didnt even think about it as the stock library uses ILI9341 with a few tweaks. It now plays video perfectly.

I have played with audio quite a bit trying to get it to work with the internal DAC (which is pretty terrible at the best of times) most of the editing needs to be done in audio task sketch. Best I got was loud garbled sound which crashed the whole unit after 5 seconds, second best was some light hissing, indicates it is outputting. After listening to the DAC output on a internet radio sketch, I dont think it'll ever be much good anyway to be honest.

**Reply by KevynWithaY** [reply]
:-) Glad the video is working. And there are hardware mods you can do to the CYDs for the DAC issues if you're so inclined (and brave, don't shake, are good with soldering teeny tiny components, etc.). I thought it would be cool (and a workaround) to have it output the audio over A2DP/Bluetooth to a receiver... tinkering with that still.
